linker/linker_soinfo.cpp

```diff
@@ -745,6 +746,22 @@ void soinfo::generate_handle() {
   g_soinfo_handles_map[handle_] = this;
 }
 
+ElfW(Addr) soinfo::memory_vaddr(ElfW(Addr) file_vaddr) const {
+  ElfW(Addr) res = file_vaddr + load_bias;
+
+  auto I = std::lower_bound(
+    rand_addr_segments.begin(), rand_addr_segments.end(),
+    file_vaddr, [](const SegmentInfo &a, ElfW(Addr) addr) {
+      return ((a.phdr_addr + a.real_size) < addr);
+    });
+  if (I != rand_addr_segments.end() &&
+      file_vaddr >= I->phdr_addr &&
+      file_vaddr < (I->phdr_addr + I->real_size))
+    res = I->real_addr + (file_vaddr - PAGE_START(I->phdr_addr));
+
+  return res;
```

Comment:
Refactor `file_vaddr < (I->phdr_addr + I->real_size)` into `(file_vaddr - I->phdr_addr) < I->real_size`? linker.cpp seems to do this in a few places.